client/my-sites/post-type-filter/index.jsx

```diff
@@ -55,69 +102,55 @@ const PostTypeFilter = React.createClass( {
 					label = this.translate( 'Drafts', {
 						context: 'Filter label for posts list'
 					} );
-					break;
-
-				case 'pending':
-					label = this.translate( 'Pending', {
-						context: 'Filter label for posts list'
-					} );
-					break;
-
-				case 'private':
-					label = this.translate( 'Private', {
-						context: 'Filter label for posts list'
-					} );
+					pathStatus = 'drafts';
 					break;
 
 				case 'future':
 					label = this.translate( 'Scheduled', {
 						context: 'Filter label for posts list'
 					} );
+					pathStatus = 'scheduled';
 					break;
 
 				case 'trash':
 					label = this.translate( 'Trashed', {
 						context: 'Filter label for posts list'
 					} );
+					pathStatus = 'trashed';
 					break;
 			}
 
-			let pathStatus;
-			if ( 'publish' !== status ) {
-				pathStatus = status;
-			}
-
 			return {
-				count,
+				count: jetpack ? null : count,
 				key: `filter-${ status }`,
 				path: [
 					'/types',
 					query.type,
 					pathStatus,
 					siteSlug
 				].filter( Boolean ).join( '/' ),
-				selected: query.status === status,
+				selected: mapPostStatus( pathStatus ) === query.status,
 				children: label
 			};
 		} ).filter( Boolean );
```

Comment:
Good call, I've been trying to avoid the `filter( Boolean )` lately in favor of `reduce`. I'll see if that'll be a better fit here.